### PR TITLE
[develop] Use full features everywhere in pickles

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR tidies up the various different implementations of 'generating derived features from true features'. This is now captured in a (private) record `Features.Full.t`, which everything can use, and which can only be constructed via 1 codepath.

This ensure that we do not have inconsistencies that will break pickles if/when features about the custom gates change, and that the circuits generated are as optimised as possible.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles_types/plonk_types.ml [hunk 1]
```diff
@@ -208,6 +387,93 @@ module Features = struct
...
+      } : _ Full.t =
+    let lookup_pattern_range_check =
+      (* RangeCheck, Rot gates use RangeCheck lookup pattern *)
+      range_check0 ||| range_check1 ||| rot
+    in
+    let lookup_pattern_xor =
+      (* Xor lookup pattern *)
+      xor
+    in
+    (* Make sure these stay up-to-date with the layouts!! *)
```
Comment (dannywillems) [inline src/lib/pickles_types/plonk_types.ml:437]: Nit: I would add a comment pointing to the layout definition, i.e. in `kimchi/src/circuits/lookup/lookups.rs`, function `lookups`. And do the same in the repository proof-systems. It helps the user to know where the user has to modify the code in the different places.

**Review by dannywillems — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles_types/plonk_types.ml [hunk 2]
```diff
@@ -1358,8 +1595,8 @@ module Messages = struct
         ~dummy_group_element:dummy ~bool
     in
     let lookup =
-      Lookup.opt_typ Impl.Boolean.typ ~uses_lookup ~lookups_per_row_4
-        ~runtime_tables ~dummy:[| dummy |]
+      Lookup.opt_typ Impl.Boolean.typ ~uses_lookup:uses_lookups
```
Comment (rbonichon) [inline src/lib/pickles_types/plonk_types.ml:1598]: Note for later: use the same names for the function parameter and for the field name wrt `uses_lookup(s)`

**Review by rbonichon — APPROVED** [review]
LGTM

**Comment by dannywillems** [comment]
I restarted the job `snark-transaction-profiler` on buildkite.

**Comment by dannywillems** [comment]
Ah I see, it is because it was targeting another branch than develop, and mina-mainnet is chosen therefore. See https://github.com/MinaProtocol/mina/blob/develop/buildkite/scripts/run-snark-transaction-profiler.sh#L11

**Comment by dannywillems** [comment]
Restarting the CI to make `snark-transaction-profiler` happy.

**Comment by dannywillems** [comment]
!ci-build-me

**Pull request merged**